test/Test/unit.jl

```diff
@@ -1,11 +1,118 @@
 @testset "Unit Tests" begin
     mock = MOIU.MockOptimizer(Model{Float64}())
+    mock.evalobjective = true
```

Comment:
@blegat I had to set this flag. Is it the correct way to do this? Should it be enabled by default???

Reply:
I am in favor of this being the default. Currently, JuMP uses `evalobjective=false` so using `evalobjective=true` by default will force JuMP to either do `mock.evalobjective=false` or rely on automatic evalutation of the objective.
@mlubin Which one do you prefer ?

Reply:
We can make `mock.evalobjective == true` the default. Parameters like this should be passed to the constructor, no manually set though.

Reply:
> Can't we just pass the objective value as an argument, otherwise it attempts to evaluate it?

In the constructor or in `mock_optimize!` ?

Reply:
> We can make `mock.evalobjective == true` the default. Parameters like this should be passed to the constructor, no manually set though.

Yes, we could have keyword arguments